integrated_channels/integrated_channel/exporters/learner_data.py

```diff
@@ -238,10 +239,22 @@ def _collect_grades_data(self, enterprise_enrollment, course_details):
         try:
             grades_data = self.grades_api.get_course_grade(course_id, username)
 
-        except HttpNotFoundError:
+        except HttpNotFoundError as error:
             # Grade not found, so we have nothing to report.
-            LOGGER.error("No grades data found for [%d]: [%s], [%s]", enterprise_enrollment.pk, course_id, username)
-            return None, None, None
+            try:
+                response_content = json.loads(error.content)
+                if response_content.get('error_code', '') == 'user_not_enrolled':
```

Comment:
What about the non-"user_not_enrolled" http errors - I think we should still log  those errors in an else clause - I think as written this will swallow those.

Reply:
It will, but now it will flood the logs again, I think we just need the `LOGGER.error` in an else clause so we *only* log if its not the `user_not_enrolled` error

Reply:
Sounds good, I'll take out the logger.info